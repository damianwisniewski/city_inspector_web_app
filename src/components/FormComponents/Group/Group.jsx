import React from 'react'
import PropTypes from 'prop-types'

import './Group.scss'

const Group = ({ name, children }) => (
	<section data-name={name} className='form-group'>
		{children}
	</section>
)

Group.propTypes = {
	name: PropTypes.string.isRequired,
	children: PropTypes.node.isRequired,
}

export default Group
