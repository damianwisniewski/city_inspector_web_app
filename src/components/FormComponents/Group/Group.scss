@import '../../../assets/styles/index';

.form-group {
	display: flex;
	flex-wrap: wrap;
	justify-content: space-around;
	border: 1px solid $gray;
	position: relative;
	margin: 10px;
	padding: 5px;
	border-radius: 3px;

	&::before {
		content: attr(data-name);
		display: block;
		position: absolute;
		padding: 0 5px;
		top: 0;
		left: 1.3em;
		transform: translateY(-60%);
		line-height: 1;
		font-size: 16px;
		color: $dark-gray;
		z-index: 10;
		background-color: $white;
		overflow: hidden;
		text-align: center;
	}

	&--column {
		flex-direction: column;
	}

	&--row {
		flex-direction: row;
	}
}
