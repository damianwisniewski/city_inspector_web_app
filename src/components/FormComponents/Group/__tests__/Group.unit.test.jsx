import React from 'react'
import { configure, shallow, ShallowWrapper } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

import Group from '../Group'

configure({ adapter: new Adapter() })

describe('<Group />', () => {
	/**
	 * @type {ShallowWrapper}
	 */
	let GroupElement
	const GroupContent = <input />
	const name = 'mockName'

	test('should render correctly Group', () => {
		GroupElement = shallow(<Group name={name}>{GroupContent}</Group>)
		expect(toJson(GroupElement)).toMatchSnapshot()
	})
})
