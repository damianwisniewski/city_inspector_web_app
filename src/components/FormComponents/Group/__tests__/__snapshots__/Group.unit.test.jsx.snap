// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Group /> should render correctly Group 1`] = `
<section
  className="form-group"
  data-name="mockName"
>
  <input />
</section>
`;
